src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -217,6 +220,9 @@ private async Task ProvisionAzureResources(
 
     private async Task ProcessResourceAsync(IConfiguration configuration, Lazy<Task<ProvisioningContext>> provisioningContextLazy, IAzureResource resource, CancellationToken cancellationToken)
     {
+        var beforeResourceStartedEvent = new BeforeResourceStartedEvent(resource, serviceProvider);
+        await eventing.PublishAsync(beforeResourceStartedEvent, cancellationToken).ConfigureAwait(false);
```

Comment:
Is this too early to do anything useful with the resource? Do we need to fire this later? I wonder if we're better off firing these events in the BicepProvisioner?

Reply:
I think if there are things that happen in the Bicep lifecycle that are interesting then we should define events for those separately and keep these events consistent across the various things doing orchestrations.